tomb-common/src/types/blockstore/car/carv1/v1block.rs

```diff
@@ -0,0 +1,105 @@
+use crate::types::blockstore::car::varint::{encode_varint_u128, read_varint_u128};
+use anyhow::Result;
+use std::io::{Read, Seek, Write};
+use wnfs::libipld::{
+    multihash::{Code, MultihashDigest},
+    Cid, IpldCodec,
+};
+
+// | 19-byte varint | x-byte Cid | x-byte content |
+#[derive(PartialEq, Debug)]
+pub(crate) struct V1Block {
+    pub varint: u128,
+    pub cid: Cid,
+    pub content: Vec<u8>,
+}
+
+impl V1Block {
+    pub fn new(content: Vec<u8>, codec: IpldCodec) -> Result<Self> {
+        // Compute the SHA256 hash of the bytes
+        let hash = Code::Sha2_256.digest(&content);
+        // Represent the hash as a V1 CID
+        let cid = Cid::new_v1(codec.into(), hash);
+        let varint = (cid.to_bytes().len() + content.len()) as u128;
+        // Create new
+        Ok(Self {
+            varint,
+            cid,
+            content,
+        })
+    }
+
+    /// Serialize the current object
+    pub(crate) fn write_bytes<W: Write>(&self, mut w: W) -> Result<usize> {
+        // Encode varint as buf
+        let varint_buf: Vec<u8> = encode_varint_u128(self.varint);
+        // Represent CID as bytes
+        let cid_buf: Vec<u8> = self.cid.to_bytes();
+        // Write all bytes
+        w.write_all(&varint_buf)?;
+        w.write_all(&cid_buf)?;
+        w.write_all(&self.content)?;
+        // Flush
+        w.flush()?;
+        // Return size
+        Ok(varint_buf.len() + cid_buf.len() + self.content.len())
+    }
```

Comment:
wnfs already has a `cbor::{encode, decode}` functionality that we use elsewhere in tomb, but varint code specifically i implemented myself because it requires byte level precision and can come in two forms. the V1 varints minimize bytes, but the V2 header varints are of deterministic length. Happy to utilize a library for this or the other reads in this snippet if one is deemed appropriate, but for the time being i believe this is sufficient.